src/plugins/data_view_editor/public/components/data_view_flyout_content_container.tsx

```diff
@@ -18,21 +18,33 @@ const IndexPatternFlyoutContentContainer = ({
   onCancel = () => {},
   defaultTypeIsRollup,
   requireTimestampField = false,
+  editData,
 }: DataViewEditorProps) => {
   const {
     services: { dataViews, notifications },
   } = useKibana<DataViewEditorContext>();
 
   const onSaveClick = async (dataViewSpec: DataViewSpec) => {
     try {
-      const indexPattern = await dataViews.createAndSave(dataViewSpec);
+      let indexPattern;
+      if (editData) {
+        const { name = '', timeFieldName, title = '' } = dataViewSpec;
+        editData.title = title;
+        editData.name = name;
+        editData.timeFieldName = timeFieldName;
```

Comment:
I assume it is expected that custom id is not editable
Should we disable it then when editing? 
![Screen Shot 2022-04-21 at 12 26 34](https://user-images.githubusercontent.com/7784120/164438403-b02925f2-be1f-4d09-9685-497155ca03aa.png)

Reply:
I believe we can edit it. @mattkime do you know?

Reply:
I'm not sure. Can you investigate and find out @shivindera? I think we should provide this option if possible , unless the SO client prevents this.